job-application-tracker-api/src/applications/dto/create-application.dto.ts

```diff
@@ -0,0 +1,23 @@
+import { IsNotEmpty, IsString } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+
+export class CreateApplicationDto {
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty({ example: 'Facebook' })
+    companyName: string;
+
+    @IsString()
+    @IsNotEmpty()
+    @ApiProperty({ example: 'coffe maker' })
+    position: string;
+
+    @ApiProperty()
+    states: any[];
```

Comment:
No need to allow states in CreateApplicationDto (make sure its allways creates as an empty array in the DB)